evm/contracts/examples/ConversionRate.sol

```diff
@@ -107,65 +132,142 @@ contract ConversionRate is ChainlinkClient, Ownable {
    */
   function transferLINK(address _recipient, uint256 _amount)
     public
-    onlyOwner
+    onlyOwner()
   {
     LinkTokenInterface link = LinkTokenInterface(chainlinkTokenAddress());
-    require(link.transfer(_recipient, _amount));
+    require(link.transfer(_recipient, _amount), "LINK transfer failed");
   }
 
   /**
-   * @dev Due to the ensureAllResponsesReceived modifier, this function
-   * will only run if all answers have been received by the _answerId and
-   * will clean up the answers mapping before running aggregation.
-   * @param _answerId The answer ID associated with the group of requests
+   * @notice Called by the owner to permission other addresses to generate new
+   * requests to oracles.
+   * @param _requester the address whose permissions are being set
+   * @param _allowed boolean that determines whether the requester is
+   * permissioned or not
    */
-  function updateRecords(uint256 _answerId)
-    private
-    ensureAllResponsesReceived(_answerId)
+  function setAuthorization(address _requester, bool _allowed)
+    public
+    onlyOwner
   {
-    updateRate(_answerId);
-    delete answers[_answerId];
+    authorizedRequesters[_requester] = _allowed;
+  }
+
+  /**
+   * @notice Called by the owner to kill the contract. This transfers all LINK
+   * balance and ETH balance (if there is any) to the owner.
+   */
+  function destroy()
+    public
+    onlyOwner()
+  {
+    LinkTokenInterface link = LinkTokenInterface(chainlinkTokenAddress());
+    transferLINK(owner, link.balanceOf(address(this)));
+    selfdestruct(owner);
   }
 
   /**
    * @dev Performs aggregation of the answers received from the Chainlink nodes.
+   * Assumes that at least half the oracles are honest and so can't contol the
+   * middle of the ordered responses.
    * @param _answerId The answer ID associated with the group of requests
    */
-  function updateRate(uint256 _answerId)
+  function updateLatestAnswer(uint256 _answerId)
     private
+    ensureMinResponsesReceived(_answerId)
     ensureOnlyLatestAnswer(_answerId)
```

Comment:
Don't we want to still answer if we get 2/3 or 3/4 oracle responses?

Reply:
Yes, we can always set the minimum to be all oracles, but in large sets it may be helpful to calculate updates sooner.